src/loophole.coffee

```diff
@@ -2,19 +2,35 @@ vm = require 'vm'
 
 exports.allowUnsafeEval = (fn) ->
   previousEval = global.eval
+  callback = -> global.eval = previousEval
+  error = false
   try
     global.eval = (source) -> vm.runInThisContext(source)
-    fn()
+    fn(callback)
```

Comment:
Calling `restorePreviousEval` (currently `callback`) twice shouldn't be harmful, right? So I'd just ditch worrying about tracking in `error` and sometimes call restore twice.